### DISCUSSION TITLE
dynamic power issue

### DISCUSSION THREAD

**Discussion by marvin42blue** [arkypita/LaserGRBL#2235] [General]
Hi there
I'm using LaserGRBL with my own engraver/plotter based on latest FluidNC - which does support dynamic laser power. When asking for $32 it responds properly $32=1 - but everytime I open a file and want to set "M4 dynamic power" LaserGRBL gives me this error message "...your engraver doesn't seem to support it"
1. any ideas why this does not work ?
2. is there a way to force LaserGRBL to not give me this error and always support dynamic power ?
thank, matthias

**Comment by StuartB4** [comment]
Have you enabled this in Settings.?
![Untitled66](https://github.com/arkypita/LaserGRBL/assets/16980026/7808ddb8-3b05-4046-8ab5-d7bc0e5edd6d)

**Reply by marvin42blue** [reply]
Yes, this setting is enabled

**Reply by arkypita** [reply]
> When asking for $32 it responds properly $32=1

What if you open this page (menu grbl, grbl configuration):

![image](https://github.com/arkypita/LaserGRBL/assets/8782035/be059606-e687-44d4-bdcc-af96702f84ad)

**Reply by marvin42blue** [reply]
I checked this yesterday, and my list is pretty much empty, there is only a "$1" entry
How does this work ? will LaserGRBL query those things ? or is the firmware supposed to deliver this data by itself ?

**Reply by arkypita** [reply]
LaserGRBL background query for configuration by send "$$" command when the machine is connected and idle.
Then the configuration is cached, until a new connection is established.
It can be refreshed also by press "refresh".

**Reply by marvin42blue** [reply]
Ok: here might be the problem : FluidNC does not support $$
see this : This is an old Grbl command that shows the $$ settings. We only support $10 in the Grbl numeric style
(see also http://wiki.fluidnc.com/en/features/commands_and_settings)
Do you think there is a way in LaserGRBLÖ to bring these things together ?

**Reply by arkypita** [reply]
workaround could be to connect one time LaserGrbl with its own grbl emulator 

![image](https://github.com/arkypita/LaserGRBL/assets/8782035/bbfd79a8-d563-4338-b2ee-d85f83ff2db5)

Then, connect, open the grbl configuration form, and set all the values to the same values you have in your FluidNC controller.

The important ones are:

![image](https://github.com/arkypita/LaserGRBL/assets/8782035/43275716-ac07-40c3-9526-0df1377a920f)

![image](https://github.com/arkypita/LaserGRBL/assets/8782035/842ebc8c-98f1-414f-8e8d-d74993d1d446)

As because your configuration cannot be refreshed from FluidNC, i guess the old one in cache is used (not sure).
